crypto/ec/ecp_nistz256.c

```diff
@@ -1357,9 +1402,109 @@ __owur static int ecp_nistz256_points_mul(const EC_GROUP *group,
     return ret;
 }
 
+#if defined(__x86_64) || defined(__x86_64__)
+__owur int ecp_nistz256_inv_mod_ord(const EC_GROUP *group,
+                                    BIGNUM *r,
+                                    const BIGNUM *x,
+                                    BN_CTX *ctx)
+{
+    /* RR = 2^512 mod ord(p256) */
+    static const BN_ULONG RR[P256_LIMBS] = {TOBN(0x83244c95,0xbe79eea2),
+                                            TOBN(0x4699799c,0x49bd6fa6),
+                                            TOBN(0x2845b239,0x2b6bec59),
+                                            TOBN(0x66e12d94,0xf3d95620)};
+    /* The constant 1 (unlike ONE that is one in Montgomery representation) */
+    static const BN_ULONG CONST_ONE[P256_LIMBS] = {TOBN(0,1),
+                                                   TOBN(0,0),
+                                                   TOBN(0,0),
+                                                   TOBN(0,0)};
+    /* expLo - the low 128bit of the exponent we use (ord(p256) - 2),
+       split into 4bit windows */
+    static const unsigned char expLo[32] = {0xb,0xc,0xe,0x6,
+                                            0xf,0xa,0xa,0xd,
+                                            0xa,0x7,0x1,0x7,
+                                            0x9,0xe,0x8,0x4,
+                                            0xf,0x3,0xb,0x9,
+                                            0xc,0xa,0xc,0x2,
+                                            0xf,0xc,0x6,0x3,
+                                            0x2,0x5,0x4,0xf};
+
+    BN_ULONG table[P256_LIMBS*15];
+    BN_ULONG out[P256_LIMBS], t[P256_LIMBS];
+    int i, ret = 0;
+    BIGNUM *tmp;
+
+    if ((BN_num_bits(x) > 256)
+        || BN_is_negative(x)) {
+        if ((tmp = BN_CTX_get(ctx)) == NULL) {
+            ECerr(EC_F_ECP_NISTZ256_INV_ORD, ERR_R_BN_LIB);
+            goto err;
+        }
+        if (!BN_nnmod(tmp, x, group->order, ctx)) {
```

Comment:
Needs a comment about violating constant-timeness? (This is fine, IIRC we do the same in the ecp_nist modules for out-of-range inputs - but worth a comment.)